docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -503,36 +516,87 @@ WHERE
 
 ### @STAR
 
-`@STAR` is used to return a list of projections for the given relation. It supports the following arguments, in this order:
+`@STAR` is used to return a set of column selections in a query.
+
+`@STAR` is named after SQL's star operator `*`, but it allows you to programmatically generate a set of column selections and aliases instead of just selecting all available columns. A query may use more than one `@STAR` and may also include explicit column selections.
+
+`@STAR` uses SQLMesh's knowledge of each table's columns and data types to generate the appropriate column list. The resulting query always `CAST`s columns to their data type in the source table.
 
-- `relation`: The relation to select star from
-- `alias` (optional): The alias of the relation
+`@STAR` supports the following arguments, in this order:
+
+- `relation`: The relation/table whose columns are being selected
+- `alias` (optional): The alias of the relation (if it has one)
 - `except` (optional): A list of columns to exclude
-- `prefix` (optional): A prefix to use for all selected columns
-- `suffix` (optional): A suffix to use for all selected columns
+- `prefix` (optional): A string to use as a prefix for all selected column names
+- `suffix` (optional): A string to use as a suffix for all selected column names
 - `quote_identifiers` (optional): Whether to quote the resulting identifiers, defaults to true
 
 For example, consider the following query:
 
 ```sql linenums="1"
 SELECT
-  @STAR(foo, bar, [c], 'baz_')
+  @STAR(relation=foo, alias=bar, except=[c], prefix='baz_', suffix='_qux')
```

Comment:
I don't think the `kwarg` syntax can be used in the context of SQL, let's double check this one.